### DISCUSSION TITLE
Potential Breach of Licensing

### DISCUSSION THREAD

**Discussion by alectrocute** [ExistentialAudio/BlackHole#374] [General]
Found this app which allegedly builds upon Blackhole’s source without proper licensing, thought I would post here so it is addressed.

https://www.gingeraudio.com/

Thanks.

**Comment by devinroth** [comment]
Thanks for bringing it to my attention.

Where does it say it’s building upon BlackHole? Obviously the driver looks exactly the same but I need some solid evidence before contacting them.

Regardless it’s clearly a copycat.

Devin Roth

**Comment by gchilds** [comment]
It looks to be written in c++, while BlachHole is c if I'm not mistaken.

How is it a copycat? People have been writing loopback audio drivers for the mac for as long as apple has been ignoring the need for them.

**Reply by alectrocute** [reply]
There’s two pieces of software being offered. Which one is C++?

**Comment by devinroth** [comment]
First off, BlackHole is a copycat of Soundflower. And currently BlackHole is the only working open source option now that both Soundflower and Jack no longer work on the latest versions of macOS. That's what makes BlackHole unique compared to any other solutions.

Yes there are a number of loopback drivers out there but none are as blatantly a copycat of any as GroundControl is of BlackHole. Take a look at the website https://www.gingeraudio.com vs https://existential.audio/blackhole/. 

Naming: GroundControl 2ch vs BlackHole 2ch, GroundControl 16ch vs BlackHole 16ch. I took this idea from Soundflower but they use (2ch) instead. Even the name GroundControl is a riff off the space theme I'm giving everything Existential Audio. 

Email List: Asking people to subscribe to download the installer. Again not an original idea of mine, but no other loopback driver does this.

Color: Black background with white text. Same deal as above. 

Copy: "Route audio between applications" vs "Route audio between apps". That was 100% original by me. If you compare the copy on the websites you'll see that the similarities are more than just a coincidence. 

Without a doubt https://www.gingeraudio.com took a look at my website and design and just riffed off of it instead of coming up with their own ideas. Not that my ideas are all original since I borrowed them from a number of sources but it's not like I just took one source like Jack Router, or Loopback, or any of the other examples out there and just tried to clone them.

None of this is illegal unless they are using some of BlackHole's source. It's just lazy.

**Comment by devinroth** [comment]
I'm gonna move this over to discussions.

**Comment by alectrocute** [comment]
A random Reddit comment which no longer exists called out the author for their license not being open source as the driver is a modified release of Blackhole. That’s why I wrote “allegedly”.

It looks like there’s two pieces of software offered, but the playback/monitoring utility looks completely unrelated. It’s highly suspicious the author couldn’t bundle these into the same installer. There’s also an unusual amount of coincidental features/quirks in the driver (uninstaller offered being most notable, and just so happen to have an Apple Silicon build too).

**Reply by devinroth** [reply]
link?

**Reply by alectrocute** [reply]
https://reddit.com/r/macapps/comments/mtzfc7/groundcontrol_free_virtual_audio_driver_for_mac/

**Reply by alectrocute** [reply]
I don’t want to discourage this author from continuing writing their companion monitoring app, but the fact the only difference between GroundControl’s driver is that they built BlackHole following a wiki step... is highly suspicious to only me?

**Reply by alectrocute** [reply]
I wanted to snoop around their driver but after coughing up my email; I never received a download link. My plans are foiled.

Edit: Did find a download link to the uninstaller though: https://www.dropbox.com/s/72z1pggftb9m5zq/UninstallGroundControl.pkg?dl=0
